### PR TITLE
BFD-1257: Added script for cert renewal

### PR DESCRIPTION
**Pull request by cbrunefearless**
**JIRA Ticket:**
[BFD-1257](https://jira.cms.gov/browse/BFD-1257)

**User Story or Bug Summary:**

User Story:

As a BFD engineer, the BFD's ops team should be alerted before our TLS certificates expire so that we have ample time to get them rotated. We should also ensure our TLS certs are valid and be alerted if not.
AC:

Configure automation to alert BFD team (specifically @bfd-engineers in slack #bfd-alerts channel) that all client and BFD certs will expire in
45 days (slack and email)
30 days (slack and email)
14 days (slack and email)
7 days (slack, email, and trigger splunk on-call alert)
Configure daily automation that verify's our TLS certs and sends an alert if invalid

---

### What Does This PR Do?

This is a script to check for expirations of certificates and alert someone depending on the amount of days it has to expire.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [ ] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by keithdadkins — DISMISSED** [review]

**Comment by christrotteradhoc** [comment]
Very minor thing but in the PR template if you replace the space inside each of the checkboxes with an 'X' (without leaving a space) it will show up nicer in here.

**Comment by christrotteradhoc** [comment]
This looks like it does not include the configuration file that will actually turn this on and make it start happening. Is that correct? If so where is that configuration going to be? Is it in another PR to come later?

File: ops/ccs-ops-misc/cert-checker/cert-checker.sh [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+# whether or not to send out notifications (defaults to false, add -s flag to enable)
+SEND_NOTIFICATIONS=${SEND_NOTIFICATIONS:-"false"}
+
+# schedule info
+CRON_SCHEDULE="${SCHEDULE:-"0 12 * * *"}" # run at noon everyday
+
+# notifications
+WARN_WHEN="${WARN_WHEN:-"3000:4500"}"     # WARN when expiration is >= 30 but less than 45 days
+ALERT_WHEN="${ALERT_WHEN:-"1500:3000"}"   # expiration is >= 15 but less than 30 days
+PAGE_WHEN="${PAGE_WHEN:-"1:1500"}"        # expiration is >= 0 but less than 15 days
```
Comment (christrotteradhoc) [inline ops/ccs-ops-misc/cert-checker/cert-checker.sh:14]: these values are very different from the sample which are '30:45', etc. Which is correct?

File: ops/ccs-ops-misc/cert-checker/cert-checker.sh [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+# TODO: check a single certificate
+# example: check_cert /path/to/cert
+check_cert() {
+  local cert="$1"
+  echo "Checking $cert"
+
+  # fail early if we cannot read the cert file
+  [[ ! -f "$1" || ! -r "$1" ]] && error_exit "cannot access $cert"
+  
+  # TODO
```
Comment (christrotteradhoc) [inline ops/ccs-ops-misc/cert-checker/cert-checker.sh:189]: is this supposed to do something?

Comment (cbrunefearless) [reply ops/ccs-ops-misc/cert-checker/cert-checker.sh:189]: JUst a todo for future cert checks

File: ops/ccs-ops-misc/cert-checker/cert-checker.sh (hunk 1 above)
Comment (cbrunefearless) [reply ops/ccs-ops-misc/cert-checker/cert-checker.sh:14]: changed

File: ops/ccs-ops-misc/cert-checker/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+To Run Script
```
Comment (keithdadkins) [inline ops/ccs-ops-misc/cert-checker/README.md:1]: Nitpicky, but this should be markdown formatted text. Try

```markdown
## To run this script:

```shell
chmod +x ./cert-checker
mv .env.sample .env
vi .env # (see ops for the correct settings)
source .env && ./cert-checker
# ``` <- add closing ticks
```

**Review by keithdadkins — CHANGES_REQUESTED** [review]

**Comment by keithdadkins** [comment]
> This looks like it does not include the configuration file that will actually turn this on and make it start happening. Is that correct? If so where is that configuration going to be? Is it in another PR to come later?

Yeah, we will need to sort out where this lives (but likely in jenkins).

File: ops/ccs-ops-misc/cert-checker/README.md (hunk 3 above)
Comment (cbrunefearless) [reply ops/ccs-ops-misc/cert-checker/README.md]: done

**Review by christrotteradhoc — APPROVED** [review]

**Review by keithdadkins — APPROVED** [review]

**Pull request merged**